x-pack/plugins/infra/public/components/metrics_explorer/chart_context_menu.test.tsx

```diff
@@ -5,103 +5,116 @@
-    expect(menuItems.length).toBe(2);
-    expect(menuItems.at(0).text()).toBe('Open in Visualize');
-    expect(menuItems.at(1).text()).toBe('View metrics for host');
-  });
+      component.find('button').simulate('click');
+      expect(getTestSubject(component, 'metricsExplorerAction-AddFilter').length).toBe(0);
+    });
 
-  it('should not display "Add Filter" without options.groupBy', async () => {
-    const customOptions = { ...options, groupBy: void 0 };
-    const onFilter = jest.fn().mockImplementation((query: string) => void 0);
-    const component = mountWithIntl(
-      <MetricsExplorerChartContextMenu
-        timeRange={timeRange}
-        source={source}
-        series={series}
-        options={customOptions}
-        onFilter={onFilter}
-      />
-    );
+    it('should not display "Add Filter" without options.groupBy', async () => {
+      const customOptions = { ...options, groupBy: void 0 };
+      const onFilter = jest.fn().mockImplementation((query: string) => void 0);
+      const component = mountWithIntl(
+        <MetricsExplorerChartContextMenu
+          timeRange={timeRange}
+          source={source}
+          series={series}
+          options={customOptions}
+          onFilter={onFilter}
+        />
+      );
 
-    component.find('button').simulate('click');
-    const menuItems = component.find('.euiContextMenuItem__text');
-    expect(menuItems.length).toBe(1);
-    expect(menuItems.at(0).text()).toBe('Open in Visualize');
-  });
+      component.find('button').simulate('click');
+      expect(getTestSubject(component, 'metricsExplorerAction-AddFilter').length).toBe(0);
+    });
+
+    it('should disable "Open in Visualize" when options.metrics is empty', async () => {
+      const customOptions = { ...options, metrics: [] };
+      const component = mountWithIntl(
+        <MetricsExplorerChartContextMenu
+          timeRange={timeRange}
+          source={source}
+          series={series}
+          options={customOptions}
+        />
+      );
 
-  it('should disable "Open in Visualize" when options.metrics is empty', async () => {
-    const customOptions = { ...options, metrics: [] };
-    const component = mountWithIntl(
-      <MetricsExplorerChartContextMenu
-        timeRange={timeRange}
-        source={source}
-        series={series}
-        options={customOptions}
-      />
-    );
+      component.find('button').simulate('click');
+      expect(
+        getTestSubject(component, 'metricsExplorerAction-OpenInTSVB').prop('disabled')
+      ).toBeTruthy();
+    });
+  });
 
-    component.find('button').simulate('click');
-    const menuItems = component.find('button.euiContextMenuItem');
-    expect(menuItems.length).toBe(1);
-    expect(menuItems.at(0).prop('disabled')).toBeTruthy();
+  describe('helpers', () => {
+    test('createNodeDetailLink()', () => {
+      const to = DateMath.parse('now', { roundUp: true });
+      const from = DateMath.parse('now-1h');
+      if (!to || !from) {
```

Comment:
If we wanted we avoid this by using the definite assignment operator in the two lines above:

```typescript
const to = DateMath.parse('now', { roundUp: true })!;
const from = DateMath.parse('now-1h')!;
```